universal-application-tool-0.0.1/app/views/admin/programs/ProgramBlockPredicatesEditView.java

```diff
@@ -105,6 +111,39 @@ private Modal predicateFormModal(
 
   private Tag renderPredicateForm(
       String blockName, QuestionDefinition questionDefinition, Tag csrfTag) {
+
+    ImmutableMap<Scalar, ScalarType> scalars;
+    try {
+      scalars = Scalar.getScalars(questionDefinition.getQuestionType());
+    } catch (InvalidQuestionTypeException | UnsupportedQuestionTypeException e) {
+      return div()
+          .withText("Sorry, you cannot create a show/hide predicate with this question type.");
+    }
+
+    ImmutableList.Builder<SimpleEntry<String, String>> scalarOptionsBuilder =
+        ImmutableList.builder();
+    scalars.forEach(
+        (scalar, type) -> {
+          scalarOptionsBuilder.add(new SimpleEntry<>(scalar.toString(), scalar.toString()));
+        });
+
+    ContainerTag valueField;
+    if (questionDefinition.getQuestionType().isMultiOptionType()) {
+      ImmutableList<SimpleEntry<String, String>> valueOptions =
```

Comment:
thought: I'm not a fan of `SimpleEntry`. I might change `SelectWithLabel` to use a map instead, or generalize to just `Entry`. Perhaps this will be my late Friday code cleanup :)

Reply:
That would be awesome!